File: services/content-service/internal/domain/announcement.go
package domain

import (
	"context"
	"time"
)

// Announcement represents an announcement entity
type Announcement struct {
	ID                  int64
	Title               string
	Content             string
	StartsAt            *time.Time
	EndsAt              *time.Time
	IsActive            bool
	IsDismissible       bool
	ShowForCustomers    bool
	ShowOnFrontend      bool
	ShowOnUserDashboard bool
	CreatedAt           *time.Time
	UpdatedAt           *time.Time
}

// AnnouncementRepository defines the interface for announcement data access
type AnnouncementRepository interface {
	GetByID(ctx context.Context, id int64) (*Announcement, error)
	GetAll(ctx context.Context, page, perPage int, search, sortBy, sortOrder string) ([]*Announcement, int64, error)
	Create(ctx context.Context, announcement *Announcement) (*Announcement, error)
	Update(ctx context.Context, announcement *Announcement) (*Announcement, error)
	Delete(ctx context.Context, id int64) error
	GetActiveAnnouncements(ctx context.Context, forCustomers, forFrontend, forUserDashboard bool) ([]*Announcement, error)
}

// AnnouncementService defines business logic for announcements
type AnnouncementService interface {
	GetAnnouncementByID(ctx context.Context, id int64) (*Announcement, error)
	GetAllAnnouncements(ctx context.Context, page, perPage int, search, sortBy, sortOrder string) ([]*Announcement, int64, error)
	CreateAnnouncement(ctx context.Context, announcement *Announcement) (*Announcement, error)
	UpdateAnnouncement(ctx context.Context, announcement *Announcement) (*Announcement, error)
	DeleteAnnouncement(ctx context.Context, id int64) error
	GetActiveAnnouncements(ctx context.Context, forCustomers, forFrontend, forUserDashboard bool) ([]*Announcement, error)
}

File: web/src/types/announcement.ts
export interface Announcement {
  id: string
  title: string
  content: string
  startsAt?: number | null
  endsAt?: number | null
  isActive: boolean
  isDismissible: boolean
  showForCustomers: boolean
  showOnFrontend: boolean
  showOnUserDashboard: boolean
  createdAt: number
  updatedAt: number
}

export interface AnnouncementsResponse {
  announcements: {
    success: boolean
    message: string
    data: {
      announcements: Array<Announcement>
      total: number
      page: number
      perPage: number
    }
  }
}

export interface AnnouncementResponse {
  announcement: {
    success: boolean
    message: string
    data: Announcement
  }
}

export interface ActiveAnnouncementsResponse {
  activeAnnouncements: {
    success: boolean
    message: string
    data: Array<Announcement>
  }
}

export interface DeleteAnnouncementResponse {
  deleteAnnouncement: {
    success: boolean
    message: string
  }
}

export interface CreateAnnouncementInput {
  title: string
  content: string
  startsAt?: number
  endsAt?: number
  isActive?: boolean
  isDismissible?: boolean
  showForCustomers?: boolean
  showOnFrontend?: boolean
  showOnUserDashboard?: boolean
}

export interface UpdateAnnouncementInput {
  id: string
  title: string
  content: string
  startsAt?: number
  endsAt?: number
  isActive?: boolean
  isDismissible?: boolean
  showForCustomers?: boolean
  showOnFrontend?: boolean
  showOnUserDashboard?: boolean
}

export interface CreateAnnouncementResponse {
  createAnnouncement: {
    success: boolean
    message: string
    data: Announcement
  }
}

export interface UpdateAnnouncementResponse {
  updateAnnouncement: {
    success: boolean
    message: string
    data: Announcement
  }
}

export interface DeleteAnnouncementResponse {
  deleteAnnouncement: {
    success: boolean
    message: string
  }
}
